### PR TITLE
[X86][MC] Support 'ctestX <reg>, <mem>' and 'ctextX <mem>, <reg>' as synonyms.

### PR DESCRIPTION
**Pull request by FreddyLeaf**
relate gas discussion: https://sourceware.org/pipermail/binutils/2024-July/135349.html

### CODE REVIEW

File: llvm/test/MC/X86/apx/ctest-att.s [hunk 1]
```diff
@@ -27,6 +27,19 @@
 # CHECK: ctestbq {dfv=of} %r9, 123(%r8,%rax,4)
 # CHECK: encoding: [0x62,0x54,0xc4,0x02,0x85,0x4c,0x80,0x7b]
          ctestbq {dfv=of} %r9, 123(%r8,%rax,4)
+# Swap mr form
+# CHECK: ctestbb {dfv=of} %bl, 123(%r8,%rax,4)
+# CHECK: encoding: [0x62,0xd4,0x44,0x02,0x84,0x5c,0x80,0x7b]
+         ctestbb {dfv=of} 123(%r8,%rax,4), %bl
+# CHECK: ctestbw {dfv=of} %dx, 123(%r8,%rax,4)
+# CHECK: encoding: [0x62,0xd4,0x45,0x02,0x85,0x54,0x80,0x7b]
```
Comment (KanRobert) [inline llvm/test/MC/X86/apx/ctest-att.s:35]: The swap form is quite magic, could you put it at the end of the test

**Comment by FreddyLeaf** [comment]
ping

**Comment by FreddyLeaf** [comment]
ping

**Review by KanRobert — APPROVED** [review]
LGTM

**Comment by KanRobert** [comment]
Please add the link of GCC patch/discussion in the description

**Pull request merged**